### ISSUE TITLE
Undocumented surprise requirement of the argocd-cm configmap

### ISSUE THREAD

**Issue by thefirstofthe300** [argoproj/argo-cd#2931]
As far as I can tell, this particular requirement is not documented anywhere but in the source code: https://github.com/argoproj/argo-cd/blob/91b0cd0a4745f288e78a8cff2dfdf942281e5e1d/util/settings/settings.go#L288

I just spent about three hours trying to figure out why Argo was not seeing a CM that was clearly there. Just wanting to help save other people some hassle.

For posterity and to make this issue more searchable, the error I saw was 

```
configmap \"argocd-cm\" not found
```

**Comment by asaushkin** [comment]
I had been trying to use argocd-util and also found this bug. But my ConfigMap is annotated with this label.

```
kind: ConfigMap
metadata:
  annotations:
  labels:
    app.kubernetes.io/name: argocd-cm
    app.kubernetes.io/part-of: argocd
  name: argocd-cm
  namespace: argocd
```

Trying to run export:

```
$ argocd-util export 
FATA[0000] configmaps "argocd-cm" not found
```

**Comment by jakehamilton** [comment]
I also just ran into this and it was quite confusing since I couldn't find anything mentioned in the documentation anywhere.

Would it be possible to make this either configurable or more explicit via annotations or custom labels?

**Comment by jfdumont** [comment]
> ```
> $ argocd-util export 
> FATA[0000] configmaps "argocd-cm" not found
> ```

I had to explicitly give the namespace
```
$ argocd-util -n argocd export 
```

**Comment by jannfis** [comment]
@thefirstofthe300 Oops, this has been around for too long. Thank you for your issue, you are totally right about this being rather undocumented (except in the release notes to v1.2 somewhere).

I have submitted a fix for the documentation, stating there is required annotation for ConfigMap resources unless we find a better way for handling the CMs.

**Comment by edtshuma** [comment]
For me what worked was switching the default namespace to argocd. Use-case was adding minikube  to argocd instance 

Set of steps 

Login to argo
```
argocd login --core
```

Switch namespace
```
kubectl config set-context --current --namespace=argocd
```

Register minikube cluster with ArgoCD
```
argocd cluster add minikube --server https://localhost:8080
```

**Comment by christianh814** [comment]
Addressed in #3540
